internal/http/services/owncloud/ocdav/tags.go

```diff
@@ -70,33 +71,57 @@ func (h *TagHandler) Handler(s *svc) http.Handler {
 }
 
 func (h *TagHandler) handleCreateTags(w http.ResponseWriter, r *http.Request, s *svc, rid *provider.ResourceId) {
-	h.modifyTags(w, r, s, rid, func(ts *tags.Tags, newtags string) bool {
+	h.modifyTags(w, r, s, rid, func(ts *tags.Tags, newtags string, info *provider.ResourceInfo) bool {
+		log := appctx.GetLogger(r.Context()).With().Interface("resourceid", rid).Logger()
+
 		if !ts.AddList(newtags) {
 			w.WriteHeader(http.StatusBadRequest)
-			log := appctx.GetLogger(r.Context()).With().Interface("resourceid", rid).Logger()
 			b, err := errors.Marshal(http.StatusBadRequest, "no new tags in createtagsrequest or maximum reached", "")
 			errors.HandleWebdavError(&log, w, b, err)
 			return false
 		}
+
+		if err := events.Publish(s.stream, events.TagsAdded{
+			Tags: newtags,
+			Ref: &provider.Reference{
+				ResourceId: rid,
+				Path:       ".",
+			},
+			Executant: info.Owner,
+		}); err != nil {
+			log.Error().Err(err).Msg("Failed to publish TagsAdded event")
+		}
+
```

Comment:
It could lead to problems to send the event within the modify func. If there is an error happening when actually setting the tags the event would still be send, even though there where no tags added.

Better: 
- Change `h.modifyTag` to take an `ev interface{}` as parameter. 
- in `h.modifyTag` if tags where successfully set and `ev != nil` => Publish(ev)`